src/index.ts

```diff
@@ -1,92 +1,166 @@
+    }
+
+    addLecturer(lecturer: string): void {
+        this._lecturers.push(lecturer);
+    }
 
-    get areas() {
-        this._areas;
+    removeLecturer(lecturer: string): void {
+        this._lecturers = this._lecturers.filter((element: string): boolean => element !== lecturer);
     }
 
-    get lecturers() {
-        this._lecturers;
+
+    get areas(): string[] {
+        return this._areas;
+    }
+
+    get lecturers(): string[] {
+        return this._lecturers;
     }
 }
 
 class Area {
     // implement getters for fields and 'add/remove level' methods
-    _levels = [];
-    _name;
+    _levels: number[] = [];
+    _name: string;
 
-    constructor(name) {
+    constructor(name: string) {
         this._name = name;
     }
+
+    addLevel(level: number): void {
+        this._levels.push(level);
+    }
+
+    removeLevel(level: number): void {
+        this._levels = this._levels.filter((element: number): boolean => element !== level);
+    }
+
+    get levels(): number[] {
+        return this._levels;
+    }
+
+    get name(): string {
+        return this._name;
+    }
 }
 
 class Level {
     // implement getters for fields and 'add/remove group' methods
+    _groups: string[] = [];
+    _name: string;
+    _description: string;
 
-    _groups;
-    _name;
-
-    constructor(name, description) {
-        this.name = name;
+    constructor(name: string, description: string) {
+        this._name = name;
         this._description = description;
     }
+
+    addGroup(group: string): void {
+        this._groups.push(group);
+    }
+
+    removeGroup(group: string) {
+        this._groups = this._groups.filter((element: string): boolean => element !== group);
+    }
+
+    get groups(): string[] {
+        return this._groups;
+    }
+
+    get name(): string {
+        return this._name;
+    }
+
+    get description(): string {
+        return this._description;
+    }
 }
 
 class Group {
     // implement getters for fields and 'add/remove student' and 'set status' methods
 
-    _area;
-    _status;
-    _students = []; // Modify the array so that it has a valid toSorted method*
+    _area: string = '';
+    _status: string = '';
+    _students: string[] = []; // Modify the array so that it has a valid toSorted method*
+    _directionName: string;
+    _levelName: string;
 
-    constructor(directionName, levelName) {
-        this.directionName = directionName;
-        this.levelName = levelName;
+    constructor(directionName: string, levelName: string) {
+        this._directionName = directionName;
+        this._levelName = levelName;
     }
 
-    showPerformance() {
-        const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+    showPerformance(): string[] {
+        const sortedStudents: string[] = this._students.toSorted((a: string, b: string) => b.getPerformanceRating() - a.getPerformanceRating());
```

Comment:
Do you have any errors in that line? It seems to have an incorrect type annotation. Please double-check it.